Source/Core/Core/IOS/Network/KD/NWC24DL.cpp

```diff
@@ -120,6 +120,11 @@ bool NWC24Dl::IsEncrypted(u16 entry_index) const
   return !!Common::ExtractBit(Common::swap32(m_data.entries[entry_index].flags), 3);
 }
 
+bool NWC24Dl::IsNotRSASigned(u16 entry_index) const
+{
+  return !!Common::ExtractBit(Common::swap32(m_data.entries[entry_index].flags), 2);
```

Comment:
I see the same for the `IsEncrypted` but do we ever check if the index is in the bounds?

Reply:
On line 327 of `Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp`, it does that for us.
```cpp
if (entry_index >= NWC24::NWC24Dl::MAX_ENTRIES)
{
    ERROR_LOG_FMT(IOS_WC24, "NET_KD_REQ: Entry index out of range.");
    LogError(ErrorType::KD_Download, NWC24::WC24_ERR_INVALID_VALUE);
    WriteReturnValue(NWC24::WC24_ERR_INVALID_VALUE, request.buffer_out);
    return IPCReply(IPC_SUCCESS);
}
```